src/tests/system/tests/test_cache.py

```diff
@@ -412,3 +416,62 @@ def test_cache__both_ldap_user_email_and_extra_attribute_email_are_stored(client
     assert user_dict["description"] == ["gecos1"], "attribute 'description' was not correct"
     assert user_dict["mail"] == ["user1@example.test"], "attribute 'mail' was not correct"
     assert user_dict["email"] == ["user1@example.test"], "attribute 'email' was not correct"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_cache__sssd_cache_nowait_percentage(client: Client, provider: GenericProvider):
+    """
+    :title: Test SSSD netgroup caching with 'entry_cache_nowait_percentage'
+    :setup:
+        1. A netgroup is created and member is added to the netgroup
+        3. SSSD is configured with entry_cache_nowait_percentage
+        4. The domain "test" is updated with `entry_cache_timeout` set to 30 seconds
+        5. SSSD is restarted with the `clean=True` option to apply the new configuration
+    :steps:
+        1. Record the initial response time for netgroup query
+        2. Introduce a network delay of 50 seconds
+        3. Execute another query for netgroup
+        4. Record the response time and verify it is faster than the initial query
+        5. Check the SSSD NSS logs for the presence of the log entry:
+            "Performing midpoint cache update of [netgrp_nowait@test]"
+    :expectedresults:
+        1. The query should return the netgroup with the correct name
+        2. Network delayed for 50 seconds
+        3. Query is executed
+        4. The response time for subsequent queries should be faster than the initial query
+        6. The expected log entries are present in the logs
+    :customerscenario: True
+    """
+    if not isinstance(provider, (LDAP, Samba, AD)):
```

Comment:
Let's change this condition to check for what it should skip, IPA.
```
if isinstance(provider, IPA):
```

Reply:
tests/test_netgroups.py:348: error: Unexpected keyword argument "domain" for "add_member" of "GenericNetgroup"  [call-arg]
.venv/lib/python3.11/site-packages/sssd_test_framework/roles/generic.py:794: note: "add_member" of "GenericNetgroup" defined here

Reply:
So this needs to be this way only

Reply:
That got me thinking. First of all, why would you use `@pytest.mark.topology(KnownTopologyGroup.AnyProvider)` if you're not going to test it with all the topologies in that group? In other tests you correctly put exactly the exact topologies on which this test is going to run:
```
@pytest.mark.topology(KnownTopology.LDAP)
@pytest.mark.topology(KnownTopology.AD)
@pytest.mark.topology(KnownTopology.Samba)
```

Reply:
changed